jsk_rviz_plugins/src/camera_info_display.cpp

```diff
@@ -405,10 +409,16 @@ namespace jsk_rviz_plugins
   {
     boost::mutex::scoped_lock lock(mutex_);
     cv_bridge::CvImagePtr cv_ptr;
+    if (!camera_info_) {
+      return;
+    }
     try
     {
       cv_ptr = cv_bridge::toCvCopy(msg, sensor_msgs::image_encodings::RGB8);
-      image_ = cv_ptr->image;
+      cv::Rect roi(camera_info_->roi.x_offset, camera_info_->roi.y_offset,
+                   camera_info_->roi.width ? camera_info_->roi.width : camera_info_->width,
+                   camera_info_->roi.height ? camera_info_->roi.height : camera_info_->height);
+      image_ = cv::Mat(cv_ptr->image, roi).clone();
```

Comment:
> When I put a check to use image and mistakenly select smaller image than camera info, 

I don't know why, but `/kinect_head/depth_downsample/camera_info` and `/robot_to_mask_image/output` also did not work. This may be wrong.

Reply:
Thanks for your review!

The image size in camera_info should be equal to the size of the image. 
So I changed the code to check those sizes and return if they are different.

> When I put a check to use image and mistakenly select smaller image than camera info, process has died with following error.

This change solved the error. Please check it.
https://github.com/jsk-ros-pkg/jsk_visualization/pull/851/commits/c786fe34a7c1ee2297f279a115f1432bfb3e34ea

Reply:
> I don't know why, but /kinect_head/depth_downsample/camera_info and /robot_to_mask_image/output also did not work. This may be wrong.

`CameraInfo` for ` /robot_to_mask_image/output` is different from `/kinect_head/depth_downsample/camera_info`.
I created a PR to publish `CameraInfo` for  ` /robot_to_mask_image/output`.
https://github.com/nakane11/jsk_recognition/pull/3

Reply:
Thanks! 
Do you think `cv::Rect roi` should have the same resolution as the original image?
If roi is devided by binning, `/kinect_head/depth_downsample/camera_info` could show downsampled image but `/kinect_head/rgb/image_rect_color` becomes unavailable.

Reply:
Thanks for your ideas.

> If roi is devided by binning, /kinect_head/depth_downsample/camera_info could show downsampled image but /kinect_head/rgb/image_rect_color becomes unavailable.

I modified the code to be able to visualize downsampled image such as `/kinect_head/depth_downsample/camera_info` with `/kinect_head/rgb/image_rect_color`.
https://github.com/jsk-ros-pkg/jsk_visualization/pull/851/commits/92d7bd08fb4abd2a5d63736293a75908484a5a6b

Reply:
Thank you. 
I confirmed that both size image can be displayed and 16bit depth is correctly converted.

Reply:
Thanks, too!